libs/react/src/lib/form/formContext.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { Children } from 'react'
+import { FormProps } from './form'
+import { validateInputValue } from './validateInput'
+
+interface FormContextReturnType {
+  setValues: React.Dispatch<React.SetStateAction<Record<string, any>>>
+  setErrors: React.Dispatch<React.SetStateAction<Record<string, any>>>
+  setFields: React.Dispatch<React.SetStateAction<Record<string, any>>>
+  errors?: Record<string, any>
+  values?: Record<string, any>
+}
+
+export const FormContext = React.createContext<FormContextReturnType>({} as FormContextReturnType)
+
+export const useFormContext = () => {
+  return React.useContext(FormContext)
+}
+
+export const FormProvider = ({ children, direction = 'vertical', formSize = 'md', onSubmit, onFormSubmit, ...props }: React.PropsWithChildren<FormProps>) => {
+  const [values, setValues] = React.useState<Record<string, any>>()
+  const [errors, setErrors] = React.useState<Record<string, any>>()
+  const [fields, setFields] = React.useState<Record<string, any>>({})
+
+  const formSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    let hasError = false
+    event.preventDefault()
+
+    Object.keys(fields).forEach((key: string) => {
+      const errorMessage: string = validateInputValue(key, fields[key], values?.[key], setErrors) as string
+      hasError = hasError || !!errorMessage
+    })
+
+    if (!hasError) {
+      onFormSubmit && onFormSubmit(values)
+    }
+  }
+
+  const onResetForm = () => {
+    setValues({})
+    setErrors({})
+  }
```

Comment:
let's be consistent with event handler naming, if the `on` prefix is decided for the project, add to all; otherwise if it isn't maybe we can use the following naming convention
> If it is a function or method, use verbs like getName, getTotalCount, setProductName and addCustomerName etc.

ie: `submitForm`, `resetForm`

Reply:
Sure will refactor it